crates/autopilot/src/infra/solvers/authenticator.rs

```diff
@@ -0,0 +1,91 @@
+use {
+    crate::{
+        domain::{self, eth},
+        infra::blockchain::{
+            contracts::{deployment_address, Contracts},
+            ChainId,
+        },
+    },
+    ethcontract::{dyns::DynWeb3, GasPrice},
+};
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Authenticator {
+    /// The authenticator contract that decides which solver is allowed to
+    /// submit settlements.
+    authenticator: contracts::GPv2AllowListAuthentication,
+    /// The safe module that is used to provide special role to EOA.
+    authenticator_role: contracts::Roles,
+    /// The EOA that is allowed to remove solvers.
+    authenticator_eoa: ethcontract::Account,
+}
+
+///  Authenticator specific addresses
+pub struct Addresses {
+    pub authenticator_eoa: eth::H256,
```

Comment:
Now that the other addresses get passed in via the `Contracts` struct this struct could also be removed and the eoa passed directly.